### PR TITLE
[Yaml] Clean some messages + add test case

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Related to #20335 on 3.2.

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/InlineTest.php [hunk 1]
```diff
@@ -401,4 +401,13 @@ public function getTestsForDump()
...
         );
     }
+
+    /**
+     * @expectedException \Symfony\Component\Yaml\Exception\ParseException
+     * @expectedExceptionMessage Malformed inline YAML string: {this, is not, yaml}.
+     */
+    public function testStringOffsetCastError()
+    {
+        Inline::parse('{this, is not, yaml}');
```
Comment (Ener-Getick) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php]: This is valid yaml, to not confuse people what about `{this, is, not, supported}`?

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -249,7 +249,7 @@ public static function parseScalar($scalar, $delimiters = null, $stringDelimiter
                 $output = $match[1];
                 $i += strlen($output);
             } else {
-                throw new ParseException(sprintf('Malformed inline YAML string (%s).', $scalar));
+                throw new ParseException(sprintf('Malformed inline YAML string: %s.', $scalar));
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Inline.php]: Why?

Comment (stof) [inline src/Symfony/Component/Yaml/Inline.php]: this avoids confusion, especially in case the YAML string contains unbalanced braces

Comment (stof) [inline src/Symfony/Component/Yaml/Inline.php]: yeah, messages without brackets were changed in the past precisely for the reason I gave. Looks like messages added in newer branches were not updated after the branch merge

**Comment by xabbuh** [comment]
👍

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**